oauthenticator/globus.py

```diff
@@ -72,22 +43,58 @@ class GlobusLoginHandler(OAuthLoginHandler, GlobusMixin):
     def get(self):
         redirect_uri = self.authenticator.get_callback_url(self)
         client = self.authenticator.globus_portal_client()
+        state = self.get_state()
+        self.set_state_cookie(state)
         client.oauth2_start_flow(
             redirect_uri,
             requested_scopes=' '.join(self.authenticator.scope),
-            refresh_tokens=self.authenticator.allow_refresh_tokens
+            refresh_tokens=self.authenticator.allow_refresh_tokens,
+            state=state
```

Comment:
`state` wasn't previously set in the OAuth flow, which began causing errors since the base authenticator started checking for it. Now that it's properly set, the OAuth flow should proceed normally.

Reply:
Yes. `oauth2_start_flow` does provide the base auth URL which would automatically update if the Globus Auth URL were ever to change, but that is unlikely any time soon and updating the URL would be trivial. I think the reduction in code is very much worth it. I'll post an update shortly.